pubtools/pulplib/_impl/fake/controller.py

```diff
@@ -204,3 +204,12 @@ def insert_task(self, task):
         """
         with self._state.lock:
             self._state.tasks.append(task)
+
+    @property
+    def repo_lock_history(self):
+        """
+        A list containing all lock actions carried out.
+        Each entry is a list itself, since several lock updates can be made
```

Comment:
In which case does this happen?  Is it only when the client decides to remove expired locks and there's more than one at a time?

It seems like a bit of a niche case, while having it be a list-of-list is more cumbersome to use than just a list, so I'm wondering if it'd be better to just flatten it.

By "more cumbersome to use", I mean for example where it has already been used in the tests:

```python
    assert len(lock_history) == 2
    assert lock_history[0][0].repository == "test-repo"
    assert lock_history[0][0].action == "lock"
    assert lock_history[1][0].repository == "test-repo"
    assert lock_history[1][0].action == "unlock"
```

This test doesn't actually verify that there were two lock-related operations.  It verifies that there were two *groups of* lock operations and it's never verified that each group contains only the expected operation (e.g. there's no `assert len(lock_history[0]) == 1`.  So that test can pass even if there are extra unexpected lock operations, and you've got to check at multiple levels of the structure if you want to avoid that.  Seems annoying to use.